hbase-server/src/main/java/org/apache/hadoop/hbase/backup/HFileArchiver.java

```diff
@@ -439,7 +439,7 @@ private static List<File> resolveAndArchive(FileSystem fs, Path baseArchiveDir,
 
     List<File> failures = new ArrayList<>();
     String startTime = Long.toString(start);
-    for (File file : toArchive) {
+    toArchive.parallelStream().forEach(file -> {
```

Comment:
There are IO operations in the code block so usually it is not safe to use parallelStream...

AFAIK parallelStream will make use of a common thread pool(ForkJoinPool.commonPool IIRC) which could be used by lots of other operations in java, so it is easy to introduce strange dead locks.

If we want to do it in parallel, we'd better introduce our own thread pool.